src/biogeophys/WaterType.F90

```diff
@@ -300,14 +301,22 @@ subroutine DoInit(this, bounds, &
     real(r8)         , intent(in) :: watsat_col(bounds%begc:, 1:)            ! volumetric soil water at saturation (porosity)
     real(r8)         , intent(in) :: t_soisno_col(bounds%begc:, -nlevsno+1:) ! col soil temperature (Kelvin)
     logical          , intent(in) :: use_aquifer_layer ! whether an aquifer layer is used in this run
+    character(len=*) , intent(in) , optional    :: NLFilename ! Namelist filename
     !
     ! !LOCAL VARIABLES:
     integer :: begc, endc
     integer :: i
+    character(len=256) :: l_NLFilename ! local for namelist filename
 
     character(len=*), parameter :: subname = 'DoInit'
     !-----------------------------------------------------------------------
 
+    !check if NLFilename is passed
```

Comment:
This can go away when you make NLFilename a required argument.

Reply:
Deleted.